lithium/lithium.py

```diff
@@ -55,7 +55,8 @@ def readTestcase(self, filename):
             # Determine whether the f has a DDBEGIN..DDEND section.
             for line in f:
                 if line.find(b"DDEND") != -1:
-                    raise LithiumError("The testcase (%s) has a line containing 'DDEND' without a line containing 'DDBEGIN' before it." % self.filename)
+                    raise LithiumError("The testcase ({}) has a line containing 'DDEND' ".format(self.filename) +
+                                       "without a line containing 'DDBEGIN' before it.")
```

Comment:
Seems like this is the difference:

```
>>> print "%s" % 1.2
1.2
>>> print "%d" % 1.2
1
>>> print "%i" % 1.2
1
```

I guess `%s` suits all purposes assuming we want to display the exact value of whatever we're passing in: "1.2"

Here's [more StackOverflow](https://stackoverflow.com/q/40714555) ascertaining the speed improvement of `%` over `.format()` as you mentioned.

Also, I may have been under the impression that the `%` modulo operator was on its way out via [PEP 3101](https://www.python.org/dev/peps/pep-3101/). Seems like due to its speed, it may still be around for a while more.